### PR TITLE
[ACS-8064] Migrate actions-favorites tests to Playwright

### PR DESCRIPTION
**Pull request by katarzynakita**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [X] Other... Please describe: Test migration

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**
Actions - Favorites tests has been migrated from Protractor to Playwright. Old Protractor tests has been deleted.
https://hyland.atlassian.net/browse/ACS-8064
Added Playwrightt's Favorite actions and missing Delete actions workflows to Github pull request run.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -174,6 +174,8 @@ jobs:
             id: 13
           - name: "upload-download-actions"
             id: 14
+          - name: "delete-actions"
```
Comment (akashrathod28) [inline .github/workflows/pull-request.yml:177]: can you please check this. there should be favorites-action e2e to be named.

File: projects/aca-playwright-shared/src/api/favorites-api.ts [hunk 2]
```diff
@@ -115,4 +115,20 @@ export class FavoritesPageApi {
       return await Utils.retryCall(favoriteFiles);
     } catch {}
   }
+
+  async removeFavoriteById(username: string, nodeId: string): Promise<void> {
+    try {
+      await this.apiService.favorites.deleteFavorite(username, nodeId);
+    } catch {}
```
Comment (akashrathod28) [inline projects/aca-playwright-shared/src/api/favorites-api.ts:122]: we can log exception message /error here.

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (akashrathod28) [reply .github/workflows/pull-request.yml:177]: Also on line 106 we can remove -"favoriteActions".

Comment (katarzynakita) [reply .github/workflows/pull-request.yml]: Ah yes, added delete too as Adam merged it yesterday

File: projects/aca-playwright-shared/src/api/favorites-api.ts [hunk 3]
```diff
@@ -115,4 +115,24 @@ export class FavoritesPageApi {
       return await Utils.retryCall(favoriteFiles);
     } catch {}
   }
+
+  async removeFavoriteById(username: string, nodeId: string): Promise<void> {
```
Comment (MichalKinas) [inline projects/aca-playwright-shared/src/api/favorites-api.ts:119]: This functions are almost the same, let's just have one to remove multiple favorites and if you need to remove one you can do it like: `[favoriteId]`

File: e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      await personalFiles.dataTable.selectItem(folder);
+      await personalFiles.acaHeader.clickMoreActions();
+      await personalFiles.matMenu.clickMenuItem('Favorite');
+      expect(await favoritesApi.isFavoriteWithRetry(username, folderId, { expect: true })).toBe(true);
+    });
+
+    test('unfavorite an item', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(fileFav1);
+      await personalFiles.acaHeader.clickMoreActions();
+      await personalFiles.matMenu.clickMenuItem('Remove Favorite');
```
Comment (MichalKinas) [inline e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts:161]: Those 3 lines look similar in several test cases, maybe they can be moved to a function?

**Comment by MichalKinas** [comment]
Also Jira ticket is not linked and PR is missing the description of changes that has been done

File: e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        await favoritesApi.addFavoriteById('folder', folderId);
+        await favoritesApi.removeFavoritesByIds(username, [fileNotFav1Id, fileNotFav2Id, fileNotFav3Id, fileNotFav4Id]);
+      } catch (error) {}
+    });
+
+    test.beforeEach(async ({ personalFiles }) => {
+      await personalFiles.dataTable.performClickFolderOrFileToOpen(parent);
+    });
+
+    test('Favorite action has empty star icon for an item not marked as favorite', async ({ personalFiles }) => {
```
Comment (akashrathod28) [inline e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts:126]: can you please add test add from test rail..

 https://hyland.atlassian.net/projects/XAT?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.xpandit.plugins.xray__testing-board&ac.selectedFolder=66311decae42b5c04ceb3fc1&ac.projectId=14544#!page=test-repository&selectedFolder=66311decae42b5c04ceb3fc1

Comment (katarzynakita) [reply e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts]: Added XRay IDs if that's what you meant :D

**Review by akashrathod28 — APPROVED** [review]

**Review by datguychen — APPROVED** [review]

**Pull request merged**